hoomd3/classic/butane/cff/butane_CFF.py

```diff
@@ -0,0 +1,303 @@
+    harmonic.params["CH"] = dict(k=2 * 340.0, r0=1.09)
+
+    angle = hoomd.md.angle.Harmonic()
+    angle.params["CCC"] = dict(k=2 * 58.35, t0=112.7 * pi / 180)
+    angle.params["CCH"] = dict(k=2 * 37.5, t0=110.7 * pi / 180)
+    angle.params["HCH"] = dict(k=2 * 33.0, t0=107.8 * pi / 180)
+
+    dihedral = hoomd.md.dihedral.OPLS()
+    dihedral.params["CCCC"] = dict(k1=1.3, k2=-0.05, k3=0.2, k4=0.0)
+    dihedral.params["HCCC"] = dict(k1=0.0, k2=0.0, k3=0.3, k4=0.0)
+    dihedral.params["HCCH"] = dict(k1=0.0, k2=0.0, k3=0.3, k4=0.0)
+
+    lj_special_pairs = hoomd.md.special_pair.LJ()
+    lj_special_pairs.params["CCCC"] = dict(epsilon=0.07, sigma=3.55)
+    lj_special_pairs.params["HCCH"] = dict(epsilon=0.03, sigma=2.42)
+    lj_special_pairs.params["HCCC"] = dict(epsilon=sqrt(0.07 * 0.03), sigma=sqrt(3.55 * 2.42))
+    lj_special_pairs.r_cut["CCCC"] = 12.0
+    lj_special_pairs.r_cut["HCCC"] = 12.0
+    lj_special_pairs.r_cut["HCCH"] = 12.0
+    coulomb_special_pairs = hoomd.md.special_pair.Coulomb()
+    coulomb_special_pairs.params["CCCC"] = dict(alpha=0.5)
+    coulomb_special_pairs.params["HCCC"] = dict(alpha=0.5)
+    coulomb_special_pairs.params["HCCH"] = dict(alpha=0.5)
+    coulomb_special_pairs.r_cut["HCCH"] = 12.0
+    coulomb_special_pairs.r_cut["CCCC"] = 12.0
+    coulomb_special_pairs.r_cut["HCCC"] = 12.0
+    integrator = hoomd.md.Integrator(dt=dt)
+    integrator.forces.append(lj)
+    integrator.forces.append(coulomb[0])
+    integrator.forces.append(coulomb[1])
+    integrator.forces.append(harmonic)
+    integrator.forces.append(angle)
+    integrator.forces.append(dihedral)
+    integrator.forces.append(lj_special_pairs)
+    integrator.forces.append(coulomb_special_pairs)
+    nvt = hoomd.md.methods.Langevin(filter=hoomd.filter.All(), kT=kT)
+    integrator.methods.append(nvt)
+    sim.operations.integrator = integrator
+
+    return sim
+
+
+# %%
+# %%
+def get_args(argv):
+    available_args = [
+        ("time-steps", "t", int, 5e5, "Number of simulation steps"),
+        ("train-freq", "f", int, 5e3, "Frequency for neural network training"),
+    ]
+    parser = argparse.ArgumentParser(description="Example script to run CFF")
+    for name, short, T, val, doc in available_args:
+        parser.add_argument("--" + name, "-" + short, type=T, default=T(val), help=doc)
+
+    return parser.parse_args(argv)
+
+
+# %%
+def main(argv=[]):
+    args = get_args(argv)
+
+    cvs = [DihedralAngle([0, 4, 7, 10])]
+    grid = Grid(lower=(-pi,), upper=(0,), shape=(64,), periodic=True)
+    topology = (8,)
+
+    method = CFF(cvs, grid, topology, kT, train_freq=args.train_freq)
+
+    timesteps = args.time_steps
+    tic = time.perf_counter()
+    run_result = pysages.run(method, generate_context, timesteps)
+    toc = time.perf_counter()
+    print(f"Completed the simulation in {toc - tic:0.4f} seconds.")
+
+    result = pysages.analyze(run_result)
+    return result["free_energy"]
```

Comment:
This script doesn't print to file the results. I think it's better to do so. And let's go for pickling (see this PR for an example: https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files)

Reply:
Plot and pickling for CFF in https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/6f6ce893013fd93698fb3637cbaf33eac2fbb4bf